common/buildcraft/factory/client/render/RenderTank.java

```diff
@@ -18,43 +25,27 @@ public RenderTank() {
                 part.sizeX = part.sizeZ = 0;
             }
             part.center.positiond(0.5, part.sizeY / 2, 0.5);
-//            TextureMap map = Minecraft.getMinecraft().getTextureMapBlocks();
-//            String flow = tile.tank.getFluidType().getFlowing().toString();
-//            TextureAtlasSprite sprite;
-//            if(map.getTextureExtry(flow) != null) {
-//                sprite = map.getTextureExtry(flow);
-//            } else {
-//                sprite = map.registerSprite(tile.tank.getFluidType().getFlowing());
-//            }
             if(tile.tank.getFluidType() == null) {
                 return;
             }
-            part.topTexture = part.bottomTexture = part.northTexture = part.southTexture = part.eastTexture = part.westTexture = tile.tank.getFluidType().getFlowing();
-            if(tile.tank.isEmpty()) {
-                part.center.colouri(0x00_FF_FF_FF);
-            } else {
-                part.center.colouri(0xFF_FF_FF_FF);
-            }
-//            part.center.colouri(0xFF_FF_FF_FF);
-//            part.center.lightf(0x1, 0);
-//            part.sizeZ = part.sizeX = 8.0 / 16.0;
-//            part.sizeY = 16.0 / 16.0;
-//            ResourceLocation endTexture = null;
-//            ResourceLocation sideTexture = null;
-//            if(tile instanceof TileMiningWell) {
-//                endTexture = ((TileMiningWell) tile).TUBE_END_TEXTURE.spriteLocation;
-//                sideTexture = ((TileMiningWell) tile).TUBE_SIDE_TEXTURE.spriteLocation;
-//            } else if(tile instanceof TilePump) {
-//                endTexture = ((TilePump) tile).TUBE_END_TEXTURE.spriteLocation;
-//                sideTexture = ((TilePump) tile).TUBE_SIDE_TEXTURE.spriteLocation;
-//            }
-//            part.topTexture = part.bottomTexture = endTexture;
-//            part.northTexture = part.southTexture = part.eastTexture = part.westTexture = sideTexture;
+            part.topTexture = part.bottomTexture = part.northTexture = part.southTexture = part.eastTexture = part.westTexture = tile.tank.getFluidType().getStill();
+            part.shouldSideBeRendered = side -> {
+                if(side.getAxis() == EnumFacing.Axis.Y) {
+                    if(Minecraft.getMinecraft().theWorld.getBlockState(tile.getPos().offset(side)).getBlock() instanceof BlockTank) {
+                        TileTank tileSide = (TileTank) Minecraft.getMinecraft().theWorld.getTileEntity(tile.getPos().offset(side));
+                        return tileSide.tank.isEmpty();
+                    }
+                    return true;
+                }
+                return true;
+            };
```

Comment:
This will probably have to be replaced with a specialised fluid renderer, as this just squishes the side texture down in odd ways. We might also need to have our own render buffer for fluid rendering, to get around the culling problem.